docs/windows.md

```diff
@@ -1,32 +1,82 @@
 
 ## What is WSL
 The Windows Subsystem for Linux lets developers run a GNU/Linux environment 
--- including most command-line tools, utilities, and applications -- 
-directly on Windows, unmodified, without the overhead of a traditional virtual 
-machine or dualboot setup.
+— including most command-line tools, utilities, and applications — 
+directly on Windows, unmodified, without the need to setup a traditional virtual 
+machine or use dualboot.
 
 ## Setup WSL
+You can either install WSL or WSL2. WSL2 has more features (notably, it includes the full Linux kernel),
+but confilcts with other virtualization software such as VirtualBox or VMware, as it requires Hyper-V to work.
+Here are the short instructions that assume installing WSL2 on a compatible Windows 10 machine running on x64.
 
-- Open PowerShell as Administrator and run:
+In case of errors, consult the full installation guide on [Microsoft Docs](https://docs.microsoft.com/en-us/windows/wsl/install-win10)
+and the [WSL Troubleshooting Guide](https://github.com/MicrosoftDocs/WSL/blob/master/WSL/troubleshooting.md).
 
-    `dism.exe /online /enable-feature /featurename:Microsoft-Windows-Subsystem-Linux /all /norestart`
+### Installing WSL2 — short guide
+All commands below should be run in an Administrator Poweshell prompt.
 
-- Enable Linux Filesystem for Windows
+- If Running Windows 10 Preview build 20262 or higher:
+    - List the available Linux Distributions
+        ```powershell
+        wsl --list --online
+        ```
+    - Install WSL2 with with a given distribution
+        ```powershell
+        wsl --install -d <distribution-name-from-above>
+        ```
+    - Restart your machine
+        ```powershell
+        Restart-Computer
+        ```
+- Else (most people):
+    - Enable WSL
+        ```powershell
+        dism.exe /online /enable-feature /featurename:Microsoft-Windows-Subsystem-Linux /all /norestart
+        ```
+    - Enable Virtual Machine Platform (required for Hyper-V and WSL2)
+        ```powershell
+        dism.exe /online /enable-feature /featurename:VirtualMachinePlatform /all /norestart
+        ```
+    - Restart your machine
+       ```powershell
+       Restart-Computer
+       ```
+    - Update WSL to WSL2
+        - Download the WSL to WSL2 updater and run it
+            ```powershell
+            (New-Object System.Net.WebClient).DownloadFile("https://wslstorestorage.blob.core.windows.net/wslblob/wsl_update_x64.msi", "C:/Windows/Temp/wsl2.msi");
+            msiexec /i "C:/Windows/Temp/wsl2.msi";
+            Remove-Item "C:/Windows/Temp/wsl2.msi"
+            ```
+        - Set WSL2 as default WSL version
+            ```powershell
+            wsl --set-default-version 2
+            ```
+    - Install a Linux distribution from the Microsoft Store (cannot be done using command line, 
+      unless using something like Chocolatey, but in that case the whole process is automated):
```

Comment:
Here we can only include distributions for which we have verified that PaSh runs (currently these are Ubuntu 18.04 and 20.04, Fedora, Debian, and Arch.)

Reply:
Ok, I removed the untested distros. Sorry, I didn't notice your comment here, but GitHub only notified me about your top-level comment.

Reply:
No worries, and thank you!